jraft-core/src/main/java/com/alipay/sofa/jraft/option/NodeOptions.java

```diff
@@ -198,6 +199,24 @@ public void validate() {
         }
     }
 
+    @Override
+    public NodeOptions copy() {
+        final NodeOptions nodeOptions = new NodeOptions();
+
+        nodeOptions.setElectionTimeoutMs(this.electionTimeoutMs);
+        nodeOptions.setSnapshotIntervalSecs(this.snapshotIntervalSecs);
+        nodeOptions.setCatchupMargin(this.catchupMargin);
+        nodeOptions.setFilterBeforeCopyRemote(this.filterBeforeCopyRemote);
+        nodeOptions.setDisableCli(this.disableCli);
+        nodeOptions.setTimerPoolSize(this.timerPoolSize);
+        nodeOptions.setCliRpcThreadPoolSize(this.cliRpcThreadPoolSize);
+        nodeOptions.setRaftRpcThreadPoolSize(this.raftRpcThreadPoolSize);
+        nodeOptions.setEnableMetrics(this.enableMetrics);
+        nodeOptions.setRaftOptions(this.raftOptions.copy());
```

Comment:
Is raftOptions exist null condition?If user set raftOptions will null object,there maybe cause NPE.

Reply:
No,it can cause NPE.You can look into NodeOptions class and RaftOptions attributs part's codes is as below:
"private RaftOptions raftOptions = new RaftOptions();"

Reply:
When call setRaftOptions method with null raftOptions,this initialization made no sense.

Reply:
Okay I will polish this codes.